### DISCUSSION TITLE
Defining a network with cubic pore and triangular throat

### DISCUSSION THREAD

**Discussion by jayaraj777** [PMEAL/OpenPNM#2749] [Q&A]
Hi,
How can I make a network with cubic pore and triangular throat. Can you help me with a function definition? I am new to coding in openPNM.  

Thanks
Jayaraj

**Comment by jgostick** [comment]
We use 'pore-scale' models for defining the behavior of pores and throats.  For instance, if you want to compute the volume of your triangular throats you'd write a function like:

``` python
def prism_volume(network):
    L = network['throat.length']
    H = network['throat.diameter']
    V = L*H/2  # Or whatever formula you like
    return V
```

You'd then add this to your network with:
``` python
network.add_model(propname='throat.volume',
                  model='prism_volume')
```

But this is the easy part.  You'll also have to do similar things for whatever physics you want to use.  For instance, for hydraulic conductance you'll need to do something like:

``` python
def gh_prism(phase, viscosity='throat.viscosity'):
    mu = phase[viscosity]
    network = phase.network
    L = network['throat.length']
    # Add the rest of your model here to compute gh
    ...
    return gh
```

If you're interested in triangular throats, then I am guessing you want to include wetting films in the corners, which adds yet another degree of difficulty.  I'm sorry, but we have not implemented these multiphase flow style conductance models, so you're on your own.

**Reply by jayaraj777** [reply]
Thanks for the quick reply.
